force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,11 +173,63 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        // Create an empty list to hold Opportunity records to be added
+        List<Opportunity> oppsToAdd = new List<Opportunity>();
+    
+        // Query to find the Account with the specified name
+        Account myAcc;
+        List<Account> accountList = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+    
+        // Check if the account was found
+        if (!accountList.isEmpty()) {
+            myAcc = accountList[0];
+        } else {
+            // If no account is found, create a new Account with the given name
+            myAcc = new Account(Name = accountName);
+            // Upsert the new Account (insert if new, update if existing)
+            upsert myAcc;
```

Comment:
This upsert should probably be outside of the else statement, it will only ever be an insert where it is here.